src/fal/planner/tasks.py

```diff
@@ -0,0 +1,159 @@
+            if model.unique_id == dbt_node:
+                model.set_status(status)
+        else:
+            model.set_status(status)
+
+
+def _map_cli_output_model_statuses(
+    run_results: dict[Any, Any]
+) -> Iterator[tuple[str, NodeStatus]]:
+    if not isinstance(run_results.get("results"), list):
+        raise Exception("Could not read dbt run results")
+
+    for result in run_results["results"]:
+        if not result.get("unique_id") or not result.get("status"):
+            continue
+
+        yield result["unique_id"], NodeStatus(result["status"])
+
+
+def _run_script(script: FalScript, fal_dbt: FalDbt):
+    print_run_info([script])
+    logger.debug("Running script {}", script.id)
+    try:
+        with _modify_path(fal_dbt):
+            script.exec(fal_dbt)
+    except:
+        logger.error("Error in script {}:\n{}", script.id, traceback.format_exc())
+        # TODO: what else to do?
+        status = FAILURE
+    else:
+        status = SUCCESS
+    finally:
+        logger.debug("Finished script {}", script.id)
+
+    return status
+
+
+@contextmanager
+def _modify_path(fal_dbt: FalDbt):
+    sys.path.append(fal_dbt.scripts_dir)
+    yield
+    sys.path.remove(fal_dbt.scripts_dir)
+
+
+@dataclass
+class DBTTask(Task):
+    model_ids: list[str]
+    _run_index: int = -1
+
+    def execute(self, args: argparse.Namespace, fal_dbt: FalDbt) -> int:
+        from fal.cli.dbt_runner import dbt_run
+
+        assert self._run_index != -1
+
+        model_names = _unique_ids_to_model_names(self.model_ids)
+        output = dbt_run(
+            args,
+            model_names,
+            fal_dbt.target_path,
+            self._run_index
+        )
+
+        for node, status in _map_cli_output_model_statuses(output.run_results):
+            _mark_dbt_nodes_status(fal_dbt, status, node)
+
+        return output.return_code
+
+
+@dataclass
+class FalModelTask(DBTTask):
+    bound_model: DbtModel | None = None
+
+    def execute(self, args: argparse.Namespace, fal_dbt: FalDbt) -> int:
+        # Run the ephemeral model
+        dbt_result = super().execute(args, fal_dbt)
+
+        # And then run the Python script if it didn't fail.
+        if dbt_result != SUCCESS:
+            return dbt_result
+
+        assert self.bound_model is not None
+        bound_script = FalScript.model_script(fal_dbt, self.bound_model)
+        script_result = _run_script(bound_script, fal_dbt=fal_dbt)
+
+        status = NodeStatus.Success if script_result == SUCCESS else NodeStatus.Error
+        _mark_dbt_nodes_status(fal_dbt, status, self.bound_model.unique_id)
+        return script_result
+
+
+@dataclass
+class FalHookTask(Task):
+    hook_path: str
+    bound_model: DbtModel | None = None
+
+    def execute(self, args: argparse.Namespace, fal_dbt: FalDbt) -> int:
+        script = FalScript(fal_dbt, self.bound_model, self.hook_path, True)
+        return _run_script(script, fal_dbt=fal_dbt)
+
+
+@dataclass
+class Node:
+    task: Task
```

Comment:
this is a `DBTTask` because `FalModelTask` extends that class. I am OK with it being `Task`, though.